src/scikit_hep_repo_review/processor.py

```diff
@@ -58,38 +59,72 @@ def build(
     fixtures: Iterable[Callable[[Traversable], Any]],
 ) -> bool | None:
     kwargs: dict[str, Any] = {}
-    signature = inspect.signature(check.check)  # type: ignore[attr-defined]
+    signature = inspect.signature(check.check)
     if "package" in signature.parameters:
         kwargs["package"] = package
 
     for func in fixtures:
         if func.__name__ in signature.parameters:
             kwargs[func.__name__] = func(package)
 
-    return check.check(**kwargs)  # type: ignore[attr-defined, no-any-return]
+    return check.check(**kwargs)
 
 
-def process(package: Traversable) -> dict[str, list[Result]]:
+def is_allowed(ignore_list: set[str], name: str) -> bool:
+    """
+    Skips the check if the name is in the ignore list or if the name without
+    the number is in the ignore list.
+    """
+    if name in ignore_list:
+        return False
+    if name.rstrip("0123456789") in ignore_list:
+        return False
+    return True
+
+
+def process(
+    package: Traversable, *, ignore: Sequence[str] = ()
+) -> dict[str, list[Result]]:
+    """
+    Process the package and return a dictionary of results.
+
+    Parameters
+    ----------
+    package: Traversable | Path
+        The Path(like) package to process
+
+    ignore: Sequence[str]
+        A list of checks to ignore
+    """
+    # Collect all installed plugins
     modules: list[str] = [
         ep.load()
         for ep in importlib.metadata.entry_points(
             group="scikit_hep_repo_review.ratings"
         )
     ]
 
+    # Collect the checks
     ratings = [
         getattr(mod, rat)
         for mod in modules
         for rat in getattr(mod, "repo_review_checks", ())
     ]
 
-    fixtures = [
+    # Collect the fixtures
+    fixtures = [pyproject] + [
         getattr(mod, fixt)
         for mod in modules
         for fixt in getattr(mod, "repo_review_fixtures", ())
     ]
 
-    tasks = {t.__name__: t for t in ratings}
+    # Collect our own config
+    config = pyproject(package).get("tool", {}).get("repo-review", {})  # type: ignore[arg-type]
+    skip_checks = set(ignore) | set(config.get("ignore", ()))
+
+    tasks: dict[str, type[Rating]] = {
+        t.__name__: t for t in ratings if is_allowed(skip_checks, t.__name__)
+    }
```

Comment:
Sorry I'm still not quite used to the tool, is it possible to maybe collect a list of `skip_checks` that do not match any existing ratings ?

Reply:
Yes, the current implementation would allow you to list anything, even if it didn't match a current check (I need to unify check/rating language).

Reply:
Currently, you can either add skips to pyproject.toml or to the args when you run the CLI tool. The skips are inspired by flake8, "D" would skip "D###` and "D123" would skip `D123`. I don't have a first digit skip ("D9" skipping "D9##`) like flake8, but Ruff doesn't really do that either, so I think this is probably fine for now. And if there's no `D*` skips in the first place, there's no warning or error. I also do not include the "family" that it came from - but that's all up for refactoring next, so not much point in adding things that might quickly be changed.

Reply:
Yeah, I think that's all good, and as you said, it is a repository to have fun, so I'm mostly thinking at loud. 

Like I use those skip in flake8, but I always find reviewing which ones are skipped to be annoying as I don't have the description. So would there be a way to in the toml file pass number and desctiption, but description be ignored ?

Like 

```
[tools.repo-review.skip]
D123="description"
W456="..."
```

(the description being ignored)


Have a flag like `repo-review --update pyproject.toml`, that would read a update pyproject.toml and inject the actual descriptions as values. 

That would make it easy to review what is skipped.

I know crazy ideas, thinking at loud.

Or maybe a toml lib support adding comments programmatically ?

Reply:
> Like

That should be supported already, I think. A list can be swapped with keys of a dict, at least in Python. It might not be supported in Ruff, but should be here. Though having a trailing comment is fine, IMO, for something ignored by the library & only for a user to read. As long as you don't have it stripped by a formatter (setup-cfg-fmt did this!), bu that's the formatter's fault. Rewriting config files is hard.

> update pyproject.toml and inject ... adding comments programmatically

Rewriting TOML without changing the format is hard. I think tomlkit can do it. I'd rather, though, have a flag that prints out something you can copy and paste into pyproject.toml, with descriptions. That would probably be pretty easy and safe.